File: src/main/java/com/EduTrack/EDUTRACK_Portfolio_MS/model/ModuleCourse.java
package com.EduTrack.EDUTRACK_Portfolio_MS.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ModuleCourse {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;

    @Enumerated(EnumType.STRING)
    private ModuleType type;

    private Double passMark;
    private String competence;
    private int credits;

    public enum ModuleType {
        GENERIC, SPECIFIC
    }
}

File: frontend/src/app/core/models/module-course.ts
export interface ModuleCourse {
    id?: number;
    type: ModuleType;
    competence: string;
    name: string;
    credits: number;
    passMark: number;
    canEqual?: boolean;
    hashCode?: boolean;
    toString?: boolean;
}

export interface ModuleCourseBuilder {
    id: number;
    name: string;
    type: ModuleType;
    competence: string;
    credits: number;
    passMark: number;
}

export enum ModuleType {
    CORE = 'CORE',
    ELECTIVE = 'ELECTIVE',
    FOUNDATION = 'FOUNDATION'
}